pkg/hubble/exporter/dynamic_exporter.go

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package exporter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	"github.com/cilium/cilium/pkg/hubble/exporter/exporteroption"
+	"github.com/cilium/cilium/pkg/hubble/metrics"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const gaugeMetricReportInterval = 5 * time.Second
+
+type dynamicExporter struct {
+	logger           logrus.FieldLogger
+	watcher          *configWatcher
+	managedExporters map[string]*managedExporter
+	maxFileSizeMB    int
+	maxBackups       int
+	mutex            lock.RWMutex
+	metricsTicker    *time.Ticker
+	metricsStop      chan bool
+}
+
+// OnDecodedEvent distributes events across all managed exporters.
+func (d *dynamicExporter) OnDecodedEvent(ctx context.Context, event *v1.Event) (bool, error) {
+	select {
+	case <-ctx.Done():
+		return false, d.Stop()
+	default:
+	}
+
+	d.mutex.RLock()
+	defer d.mutex.RUnlock()
+	for _, me := range d.managedExporters {
+		if me.config.End.After(time.Now()) {
+			me.exporter.OnDecodedEvent(ctx, event)
+		}
+	}
+	return false, nil
+}
+
+// Stop stops configuration watcher, metrics propagation and all managed flow
+// log exporters.
+func (d *dynamicExporter) Stop() error {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	d.watcher.Stop()
+	d.metricsTicker.Stop()
+	d.metricsStop <- true
+
+	var accErrs []error
+	for _, me := range d.managedExporters {
+		accErrs = append(accErrs, me.exporter.Stop())
+	}
+
+	return errors.Join(accErrs...)
+}
+
+// NewDynamicExporter creates instance of dynamic hubble flow exporter.
+func NewDynamicExporter(configFilePath string, maxFileSizeMB, maxBackups int) *dynamicExporter {
+	dynamicExporter := &dynamicExporter{
+		logger:           logrus.New(),
```

Comment:
Usually logger is passed in as a parameter.

Reply:
fixed